docs/src/main/asciidoc/credentials-provider.adoc

```diff
@@ -0,0 +1,194 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using a Credentials Provider
+
+include::./attributes.adoc[]
+:extension-status: preview
+:vault-datasource-guide: link:vault-datasource[Vault Datasource guide]
+
+Interacting with a datastore typically implies first connecting using credentials.
+Those credentials will allow the client to be identified, authenticated and eventually authorized.
+Username/password based authentication is very common, but that is not by far the only one.
+Such credentials information may appear in the application configuration,
+but it is becoming increasingly popular to store this type of sensitive information in secure stores,
+such as HashiCorp Vault, Azure Key Vault or the AWS Secrets Manager to name just a few.
+
+To bridge datastores that consume credentials, which can take different forms,
+and secure stores that provide those credentials,
+Quarkus introduces a middle abstraction called `Credentials Provider`,
+that some extensions may support to consume credentials (e.g. `agroal`),
+and some others may implement to produce credentials (e.g. `vault`).
+
+This Service Programming Interface (SPI) may also be used by implementers that want to support custom providers
+not yet  implemented in Quarkus (e.g. Azure Key Vault ).
+
+Currently, the `Credentials Provider` interface is implemented by the `vault` extension, and is supported
+by the following credentials consumer extensions:
+
+* `agroal`
+* `reactive-mysql-client`
+* `reactive-pg-client`
+
+All extensions that rely on username/password authentication do provide also the possibility to set configuration
+properties in the `application.properties` as an alternative. But the `Credentials Provider` is the only option
+if credentials are generated (e.g. `Vault Dynamic DB Credentials`) or if a custom credentials provider is required.
+
+This guide will show how to use the `Credentials Provider` provided in the `vault` extension,
+then we will look at implementing a custom `Credentials Provider`, and finally we will talk about additional
+considerations regarding implementing a `Credentials Provider` in a new extension.
+
+include::./status-include.adoc[]
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* roughly 20 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven {maven-version}
+* Docker installed
+
+== Vault Credentials Provider
+
+To configure a `Vault Credentials Provider` you need to provide the following property(ies):
+
+[source, properties]
+----
+quarkus.vault.credentials-provider.<name>.<property>=<value>
+----
+
+The `<name>` will be used in the consumer to refer to this provider. The `<property>` and `<value>` fields are specific to the `Vault Credentials Provider`. For complete details, please refer to the {vault-datasource-guide}.
```

Comment:
Does `{vault-datasource-guide}`  work? 

I'm not an adoc expert, so just asking :)

Reply:
Got it, thanks